lib/spack/spack/filesystem_view.py

```diff
@@ -67,7 +67,11 @@ def view_copy(src, dst, view, spec=None):
 
     Use spec and view to generate relocations
     """
-    shutil.copy2(src, dst)
+    if os.path.isdir(src):
+        shutil.copytree(src, dst, symlink=True)
```

Comment:
I think this needs to by `symlinks` rather than symlink? Oddly I don't see it failing, which makes me wonder, but that's how it's documented.

```suggestion
        shutil.copytree(src, dst, symlinks=True)
```